File: Python/LAKI1/H5/hello_you.py
# TIEP1.1 Lausekielinen ohjelmointi I
# Harjoitus 5, tehtävä 6
# Eveliina Toivanen (eveliina.toivanen@tuni.fi)

# Ohjelma tervehtii käyttäjää tämän antamalla nimellä.


def moikkaa(nimi):
    """Funktion parametrina on käyttäjän antama nimi.
    Funktio tulostaa ohjelman tervehdyksen.
    """
    print("Hello, ", nimi, "!", sep="")


def main():
    """"Pääohjelma tulostaa alutervehdyksen ja sen jälkeen kysyy käyttäjältä tämän
    nimeä, jonka se lähettää moikka-funktiolle parametrina.
    """

    print("Hello! I shall say hello to you.")
    print("Please, enter your name:")
    nimi = str(input())

    moikkaa(nimi)


# Kutsutaan pääohjelmaa, jos ohjelmaa ajetaan.
if __name__ == "__main__":
    main()

File: Java/Laki2/H3/HelloYou.java
/* TIEP5.1 Lausekielinen ohjelmointi II
 * Harjoitus 3, tehtävä 1
 * Eveliina Toivanen (eveliina.toivanen@tuni.fi)
 *
 * Pyydetään käyttäjältä nimi ja tulostetaan tervehdys, jossa nimi on sisällytettynä.
 */
 
import java.util.Scanner;

public class HelloYou {
    
    /* Tulostetaan tervehdys käyttäjän syöttämän nimen kanssa.
     * nimi: Käyttäjän antama nimi
     */
    public static void moikkaa(String nimi) {
        System.out.println("Hello, " + nimi + "!");
    }
    
    public static void main(String[] args){
        Scanner lukija = new Scanner(System.in);
        
        System.out.println("Hello! I shall say hello to you.");
        System.out.println("Please, enter your name:");
        String nimi = lukija.nextLine();
        
        moikkaa(nimi);
    }
}

